src/collections/queue/mod.rs

```diff
@@ -0,0 +1,80 @@
+/// A queue-like data structure implement through Vector
+pub struct Queue<T> {
+    items: Vec<T>,
+}
+
+impl<T> Queue<T> {
+    /// Initialize a queue with empty vector
+    pub fn new() -> Self {
+        Self {
+            items: Vec::new(),
+        }
+    }
+
+    /// Adds an element into queue
+    pub fn enqueue(&mut self, item: T) -> bool {
+        self.items.push(item);
+        true
```

Comment:
Assumed that allocation of underlying `Vec` would never fail (actually it may fails 😂), and also, this implementation cannot prevent [panic when the capacity exceeds the `isize::MAX`](http://doc.rust-lang.org/1.48.0/std/vec/struct.Vec.html#method.push). So that we do not need to return `true` here IMO.